### PR TITLE
Fix: also use custom private boost icon for detailed status

### LINKED ISSUE #14455: Retoot icon with lock shows when hovering on "no retoot" icon of other's followers-only toots.
### Expected behaviour 

A no retoot icon should be shown under "Followers-only" toots from other people after v3.2.0.

### Actual behaviour

When hovering on this icon, a retoot with lock icon shows itself. It's actually confusing since this icon shows under my own follower-only toots as well.

gif

### Specifications

v3.2.0

Firefox

### PR DESCRIPTION
**Pull request by OmmyZhang**
#14380
If the custom icon for private boosts is accepted, it should be used everywhere.

By the way, this change can fix #14455

### CODE REVIEW

File: app/javascript/mastodon/components/icon_button.js [hunk 1]
```diff
@@ -20,6 +20,7 @@ export default class IconButton extends React.PureComponent {
     style: PropTypes.object,
     activeStyle: PropTypes.object,
     disabled: PropTypes.bool,
+    privat: PropTypes.bool,
```
Comment (Sasha-Sorokin) [inline app/javascript/mastodon/components/icon_button.js:23]: You probably meant `private`. And you really shouldn't really pass this to shared IconButton component, there's a class name for that I believe.

**Review by brawaru — CHANGES_REQUESTED** [review]

**Comment by Sasha-Sorokin** [comment]
How does that affect #14455? Also, ThibG has already created a PR to fix it — #14456.

File: app/javascript/mastodon/components/icon_button.js (hunk 1 above)
Comment (OmmyZhang) [reply app/javascript/mastodon/components/icon_button.js:23]: No, `private` is a reserved word in JS.

It's similar to passing `disabled` as a class name. I believe this make the code structure more clear.

**Comment by OmmyZhang** [comment]
Because now custom icon is only used for `button.icon-button.privat` , and will not affect `button.icon-button.disabled`.

I have seen ThibG's PR, but my PR is mainly for using custom private boost icon everywhere.

**Comment by OmmyZhang** [comment]
diff
master
-------------------
fixed

**Review by ClearlyClaire — CHANGES_REQUESTED** [review]
Sorry for the late review! I concur with @Sasha-Sorokin, `privat` doesn't sound right. Also, I think it is far too specific to the boost icon to be part of `IconButton`'s attributes, and should use `className` instead.

**Comment by OmmyZhang** [comment]
@Sasha-Sorokin @ThibG 
Thanks for your review! Is it a little better now?

**Review by ClearlyClaire — APPROVED** [review]

**Pull request merged**